### PR TITLE
create lint-pr-title action

### PR DESCRIPTION
**Pull request by yduartep**
This PR creates a new GH action that check if the PR title match the commitlint specification based on the rules defined on the `commitlint.config.js` file.

- Added an optional `config-path` input parameter that can be filled with the absolute path of the `commitlint.config.js` file. If no value set, the action will use the file defined locally under `/actions/lint-pr-title/commitlint.config.js`.
- Added to `dependabot.yml` workflow, automatic dependencies update on `/actions/lint-pr-title` directory.
- Created a new `build-lint-pr.yml `workflow that get triggered when the dependabot PR get merged. This workflow install the dependices and build the code to update the dist/index.js file content.
- Added `@vercel/ncc` to build the project and the dependencies used and avoid storing the `node_modules` folder on the repo.

Currently, this action will be used on `hg-cloud-home-admin` and `grafana-aws-app` projects and the idea is that other projects can use it as well.

### CODE REVIEW

File: actions/lint-pr-title/dist/commit.hbs
[the remark this answers is not in the document]
Comment (yduartep) [reply actions/lint-pr-title/dist/commit.hbs:1]: no, maybe I can remove it after build

File: actions/lint-pr-title/commitlint.config.js
[the remark this answers is not in the document]
Comment (yduartep) [reply actions/lint-pr-title/commitlint.config.js:1]: I have thought about the possibility that the configuration can be optionally passed by parameter but I need to investigate what is the best option.

File: actions/lint-pr-title/package.json [hunk 1]
```diff
@@ -0,0 +1,25 @@
+{
+  "name": "lint-pr-title",
+  "version": "0.0.1",
+  "description": "Ensure your PR title matches the commit lint specification.",
+  "main": "index.js",
+  "scripts": {
+    "build": "yarn && ncc build index.js --license licenses.txt"
```
[the remark this answers is not in the document]
Comment (zerok) [reply actions/lint-pr-title/package.json:7]: Once thing that would be awesome if dependabot already ran the build step then. Haven't investigated that yet, though 🙂

**Comment by zzehring** [comment]
Looks like there's a few OSS GH Actions for `commitlint` already: 
* https://github.com/wagoid/commitlint-github-action
* https://github.com/JulienKode/pull-request-name-linter-action

Would these work for your purposes? We could reduce the amount of stuff we have to manage by using an already existing action. Then, we can just offer a reusable workflow that uses this action with an opinionated config.

**Comment by yduartep** [comment]
> Looks like there's a few OSS GH Actions for `commitlint` already:
> 
> * https://github.com/wagoid/commitlint-github-action (seems to be the more mature+currently maintained project)
> * https://github.com/JulienKode/pull-request-name-linter-action
> 
> Would these work for your purposes? We could reduce the amount of stuff we have to manage by using an already existing action. Then, we can just offer a reusable workflow that uses this action with an opinionated config.

- I was using `https://github.com/dreampulse/action-lint-pull-request-title` in other projects but the scope is mandatory. I would like to add the possibility to use different commitlint config file defined by users and this is not possible on this specific action. 

- [Lints Pull Request commits with](https://github.com/wagoid/commitlint-github-action) is not what I want. I want to check if the PR title match the specificaction not the commits on the PR.

- https://github.com/JulienKode/pull-request-name-linter-action: this is closer to what i want to achieve but you need to install @commitlint/config-conventional dependency before to use the action. I prefer to delegate all work to the action itself

File: actions/lint-pr-title/package.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (yduartep) [reply actions/lint-pr-title/package.json]: @zerok I added to `dependabot.yml` workflow, automatic dependencies update on` /actions/lint-pr-title` directory and also created a new `build-lint-pr.yml` workflow that get triggered when the dependabot PR get merged. This workflow install the dependencies and build the code to update the `dist/index.js` file content.

[the remark this answers is not in the document]
Comment (zerok) [reply actions/lint-pr-title/package.json]: LGTM, thank you :) If more JS Actions are added, we might start to unify all this but for now this is great :)

**Review by zerok — APPROVED** [review]

**Pull request merged**